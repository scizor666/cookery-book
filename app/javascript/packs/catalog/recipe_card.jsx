import React from 'react';
import {
    Card, CardImg, CardText, CardBlock,
    CardTitle, CardFooter
} from 'reactstrap';
import {confirmRecipeDelete} from '../confirm/confirm';

export default class RecipeCard extends React.Component {

    recipeLink() {
        return `${location.href}/recipe/${this.props.recipe.id}`;
    }

    handleDelete() {
        confirmRecipeDelete(this.props.recipe);
    }

    render() {
        return (<Card className="h-100">
            <CardImg top width="100%"
                src={this.props.recipe.image_url || this.props.placeholderUrl}
                alt="Card image cap"
                onClick={() => location.href = this.recipeLink()}/>
            <CardBlock>
                <CardTitle>{this.props.recipe.name}</CardTitle>
                <CardText>{this.props.recipe.short_description}</CardText>
            </CardBlock>
            <CardFooter className="text-center">
                <a title="Recipe" className="mr-5" href={this.recipeLink()}><i className="fa fa-eye"/></a>
                <a title="Edit" className="mr-5" href={this.recipeLink() + '/edit'}><i className="fa fa-pencil"/></a>
                <a title="Delete" href="javascript:void(0);" onClick={() => this.handleDelete()}
                    id={`delete_recipe_${this.props.recipe.id}`}>
                    <i className="fa fa-trash-o"/></a>
            </CardFooter>
        </Card>);
    }
}

