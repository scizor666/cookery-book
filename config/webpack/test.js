// Note: You must restart bin/webpack-dev-server for changes to take effect

const merge = require('webpack-merge')
const sharedConfig = require('./shared.js')
const enzymeConfig = require('./enzyme_exports.js')

module.exports = merge(sharedConfig, enzymeConfig)
