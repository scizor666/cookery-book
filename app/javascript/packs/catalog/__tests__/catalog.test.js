import React from 'react';
import renderer from 'react-test-renderer';
import {mount} from 'enzyme';
import {recipes} from './recipe_card.test';
import RecipeCard from '../recipe_card';
jest.mock('react-infinite-scroller', () => 'InfiniteScroll');
import Catalog from '../catalog';


test('display when loading', () => {
    const wrapper = renderer.create(<Catalog/>);
    expect(wrapper.toJSON()).toMatchSnapshot();
});

test('display a catalog with products', () => {
    const wrapper = mount(<Catalog/>);
    wrapper.setState({recipes});
    expect(wrapper.find(RecipeCard).length).toBe(2);
});