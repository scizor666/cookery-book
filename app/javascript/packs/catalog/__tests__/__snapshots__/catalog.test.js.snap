// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`display a recipe card with an image 1`] = `
<div
  className="h-100 card"
>
  <img
    alt="Card image cap"
    className="card-img-top"
    onClick={[Function]}
    src="http://some.url/with_picture.png"
    width="100%"
  />
  <div
    className="card-block"
  >
    <h4
      className="card-title"
    >
      some recipe
    </h4>
    <p
      className="card-text"
    >
      some short description here
    </p>
  </div>
  <div
    className="text-center card-footer"
  >
    <a
      className="mr-5"
      href="about:blank/recipe/123456"
      title="Recipe"
    >
      <i
        className="fa fa-eye"
      />
    </a>
    <a
      className="mr-5"
      href="about:blank/recipe/123456/edit"
      title="Edit"
    >
      <i
        className="fa fa-pencil"
      />
    </a>
    <a
      href="javascript:void(0);"
      id="delete_recipe_123456"
      onClick={[Function]}
      title="Delete"
    >
      <i
        className="fa fa-trash-o"
      />
    </a>
  </div>
</div>
`;

exports[`display a recipe card without an image 1`] = `
<div
  className="h-100 card"
>
  <img
    alt="Card image cap"
    className="card-img-top"
    onClick={[Function]}
    src="http://some.placeholder.url/with_image.png"
    width="100%"
  />
  <div
    className="card-block"
  >
    <h4
      className="card-title"
    >
      some other recipe
    </h4>
    <p
      className="card-text"
    >
      some other short description here
    </p>
  </div>
  <div
    className="text-center card-footer"
  >
    <a
      className="mr-5"
      href="about:blank/recipe/678990"
      title="Recipe"
    >
      <i
        className="fa fa-eye"
      />
    </a>
    <a
      className="mr-5"
      href="about:blank/recipe/678990/edit"
      title="Edit"
    >
      <i
        className="fa fa-pencil"
      />
    </a>
    <a
      href="javascript:void(0);"
      id="delete_recipe_678990"
      onClick={[Function]}
      title="Delete"
    >
      <i
        className="fa fa-trash-o"
      />
    </a>
  </div>
</div>
`;

exports[`display when loading 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-md-3 col-md-offset-3 container"
    >
      <div
        className="text-center"
      >
        <h1>
          My Recipes
        </h1>
        <a
          href="about:blank/recipe/new"
        >
          Add a new Recipe
        </a>
      </div>
    </div>
  </div>
  <InfiniteScroll
    hasMore={true}
    loadMore={[Function]}
    loader={
      <div
        className="loader"
      >
        Loading ...
      </div>
    }
    pageStart={0}
  >
    <div
      className="row"
    />
  </InfiniteScroll>
</div>
`;
